### PR TITLE
Create external symbol factory reexports

### PR DESCRIPTION
**Pull request by kyliau**
Exposes and fixes bug in https://github.com/angular/angular/issues/29454

Upstream PR: https://github.com/angular/angular/pull/29449

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Review by gregmagolan — APPROVED** [review]
👌

File: integration/bazel/src/BUILD.bazel [hunk 1]
```diff
@@ -54,9 +57,12 @@ rollup_bundle(
     entry_point = "src/main",
     deps = [
         "//src",
-        "@npm//@angular/common",
-        "@npm//@angular/core",
-        "@npm//@angular/platform-browser",
+        # TODO(kyliau): These are not necessary. Bundle compiles fine without
```
Comment (gregmagolan) [inline integration/bazel/src/BUILD.bazel]: Any idea why that is the case?

Comment (kyliau) [reply integration/bazel/src/BUILD.bazel]: They are direct dependencies of `@angular/router`, and it seems all transitive deps of targets listed here are included in the compilation.
I think the right thing to do here is to list them all explicitly.

Comment (kyliau) [reply integration/bazel/src/BUILD.bazel]: Specifically, it's any deps that has NodeModuleInfo provider

Comment (alexeagle) [reply integration/bazel/src/BUILD.bazel]: I think we should assume that all scripts come in from transitive runfiles and work to boil this list down to the minimal set that we have a good reason to import here

**Review by alexeagle — APPROVED** [review]
awesome!!

**Pull request closed without merging**